tasks/task_11_CSG_shut_down_dose_tallies/3_mesh_based_shut_down_dose_rate_example.py

```diff
@@ -0,0 +1,369 @@
+    dimension=[10, 10, 10],  # 10 voxels in each axis direction (x, y, z)
+)
+
+# adding a dose tally on a regular mesh
+# AP, PA, LLAT, RLAT, ROT, ISO are ICRP incident dose field directions, AP is front facing
+energies, pSv_cm2 = openmc.data.dose_coefficients(particle="photon", geometry="AP")
+dose_filter = openmc.EnergyFunctionFilter(
+    energies, pSv_cm2, interpolation="cubic"  # interpolation method recommended by ICRP
+)
+particle_filter = openmc.ParticleFilter(["photon"])
+mesh_filter = openmc.MeshFilter(mesh_photon)
+dose_tally = openmc.Tally()
+dose_tally.filters = [mesh_filter, dose_filter, particle_filter]
+dose_tally.scores = ["flux"]
+dose_tally.name = "photon_dose_on_mesh"
+
+my_gamma_tallies = openmc.Tallies([dose_tally])
+
+cells = model_neutron.geometry.get_all_cells()
+
+# # this section makes the photon sources from each active material at each
+# # timestep and runs the photon simulations
+results = openmc.deplete.Results(statepoints_folder / "neutrons" / "depletion_results.h5")
+
+for i_cool in range(1, len(timesteps)):
+
+    # we can loop through the materials in each step
+    # from the material ID we can get the mesh voxel id
+    # then we can make a MeshSource
+    # https://docs.openmc.org/en/develop/pythonapi/generated/openmc.MeshSource.html
+
+    # range starts at 1 to skip the first step as that is an irradiation step and there is no
+    # decay gamma source from the stable material at that time
+    # also there are no decay products in this first timestep for this model
+
+        photon_sources_for_timestep = []
+        strengths_for_timestep = []
+        print(f"making photon source for timestep {i_cool}")
+
+
+        mat_number_offset
+
+        step = results[i_cool]
+        activated_mat_ids = step.volume.keys()
+        print(activated_mat_ids)
+        for activated_mat_id in activated_mat_ids:
+            # gets the energy and probabilities for the 
+            activated_mat = step.get_material(activated_mat_id)
+            energy = activated_mat.get_decay_photon_energy(
+                clip_tolerance = 1e-6,  # cuts out a small fraction of the very low energy (and hence negligible dose contribution) photons
+                units = 'Bq',
+            )
+            strength = energy.integral()
+
+            if strength > 0.:  
+                source = openmc.IndependentSource(
+                    energy=energy,
+                    particle="photon",
+                    strength=strength
+                )
+            else:
+                source = openmc.IndependentSource() # how to make an empty source, source strength is set to 0
+
+            photon_sources_for_timestep.append(source)
+            strengths_for_timestep.append(strength)
+        
+        reshaped_photon_source_of_timestep = np.array(photon_sources_for_timestep).reshape(regular_mesh.dimension)
+        mesh_source = openmc.MeshSource(
+            regular_mesh, reshaped_photon_source_of_timestep
+        )
```

Comment:
no longer need to reshape thanks to another another of Pauls useful [PRs](https://github.com/openmc-dev/openmc/pull/2980)

```suggestion
        mesh_source = openmc.MeshSource(
            regular_mesh, photon_sources_for_timestep
        )
```